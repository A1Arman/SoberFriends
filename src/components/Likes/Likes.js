import React, { useState, useEffect } from 'react';
import config from '../../config';
import TokenService from '../../services/token-service';

const {API_BASE_URL} = config;

function useMergeState(initialState) {
    const [state, setState] = useState(initialState);
    const setMergedState = newState => 
        setState(prevState => Object.assign({}, prevState, newState)
    );
    return [state, setMergedState];
}

function Likes(props) {
    const [likesRequest, setLikes] = useMergeState({
        likes: null
    });

    async function fetchLikes() {
        const postId = props.postId
        const options = {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        }
        try {
            const likeResponse = await fetch(`${API_BASE_URL}/posts/${postId}/likes`, options);
            const likeData = await likeResponse.json();
            setLikes({
                likes: likeData
            })
        } catch (err) {
            throw new Error(err)
        }
    }

    useEffect(() => {
        fetchLikes();
    }, []);

    const { likes } = likesRequest;
    
    return (
        <div>
            <h4>Likes: {likes ? likes.length : 0}</h4>
        </div>
    )
}

export default Likes;