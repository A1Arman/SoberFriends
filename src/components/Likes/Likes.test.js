import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Likes from '../Likes/Likes';


describe(`Likes component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Likes />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});