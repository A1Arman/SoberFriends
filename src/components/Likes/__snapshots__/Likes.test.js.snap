// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Likes component renders without crashing 1`] = `
<div>
  <h4>
    Likes: 
    0
  </h4>
</div>
`;
